File: src/main/java/org/mifos/fineract/services/AdhocQueryApiApi.java
package org.mifos.fineract.services;

import retrofit2.http.*;
import rx.Observable;

public interface AdhocQueryApiApi {
    /**
     * @param body (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @POST("adhocquery")
    Observable<String> createAdHocQuery(
            @retrofit2.http.Body String body
    );

    /**
     * @param adHocId adHocId (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @DELETE("adhocquery/{adHocId}")
    Observable<String> deleteAdHocQuery(
            @retrofit2.http.Path("adHocId") Long adHocId
    );

    /**
     * @param adHocId adHocId (required)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("adhocquery/{adHocId}")
    Observable<String> retrieveAdHocQuery(
            @retrofit2.http.Path("adHocId") Long adHocId
    );

    /**
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("adhocquery")
    Observable<String> retrieveAll();


    /**
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @GET("adhocquery/template")
    Observable<String> template();


    /**
     * @param adHocId adHocId (required)
     * @param body    (optional)
     * @return Observable&lt;String&gt;
     */
    @Headers({
            "Content-Type:application/json"
    })
    @PUT("adhocquery/{adHocId}")
    Observable<String> update(
            @retrofit2.http.Path("adHocId") Long adHocId, @retrofit2.http.Body String body
    );

}

File: src/main/kotlin/org/openapitools/client/apis/AdhocQueryApiApi.kt
package org.openapitools.client.apis

import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.PUT
import retrofit2.http.Path


interface AdhocQueryApiApi {
    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @POST("v1/adhocquery")
    suspend fun createAdHocQuery(@Body body: String? = null): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param adHocId adHocId
     * @return [kotlin.String]
     */
    @DELETE("v1/adhocquery/{adHocId}")
    suspend fun deleteAdHocQuery(@Path("adHocId") adHocId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param adHocId adHocId
     * @return [kotlin.String]
     */
    @GET("v1/adhocquery/{adHocId}")
    suspend fun retrieveAdHocQuery(@Path("adHocId") adHocId: Long): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/adhocquery")
    suspend fun retrieveAll2(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @return [kotlin.String]
     */
    @GET("v1/adhocquery/template")
    suspend fun template(): String

    /**
     *
     *
     * Responses:
     *  - 0: default response
     *
     * @param adHocId adHocId
     * @param body  (optional)
     * @return [kotlin.String]
     */
    @PUT("v1/adhocquery/{adHocId}")
    suspend fun update(
        @Path("adHocId") adHocId: Long,
        @Body body: String? = null
    ): String

}
